src/unitxt/templates.py

```diff
@@ -360,7 +360,8 @@ def outputs_to_target_and_references(self, outputs: Dict[str, object]) -> List[s
             )
 
         if self.random_reference:
-            target = get_random().choice(references)
+            random_generator = new_random_generator(str(references))
```

Comment:
Do we want dependency on the references or the entire instance?

Reply:
updated so it depends on the entire `outputs`. is this what you meant?